Warnings in pioc_support.c for make distcheck

When doing make distcheck I see these warnings:

```
/bin/bash ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../../../src/clib -I../..   -I/usr/local/hdf5-1.10.1_mpich-3.2/include -I/usr/local/netcdf-c-4.4.1.1_mpich-3.2/include -I/usr/local/pnetcdf-1.9.0_mpich-3.2/include  -g -O2 -MT pioc_support.lo -MD -MP -MF .deps/pioc_support.Tpo -c -o pioc_support.lo ../../../../src/clib/pioc_support.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../../../src/clib -I../.. -I/usr/local/hdf5-1.10.1_mpich-3.2/include -I/usr/local/netcdf-c-4.4.1.1_mpich-3.2/include -I/usr/local/pnetcdf-1.9.0_mpich-3.2/include -g -O2 -MT pioc_support.lo -MD -MP -MF .deps/pioc_support.Tpo -c ../../../../src/clib/pioc_support.c  -fPIC -DPIC -o .libs/pioc_support.o
../../../../src/clib/pioc_support.c: In function ‘PIOc_readmap’:
../../../../src/clib/pioc_support.c:865:9: warning: ignoring return value of ‘fscanf’, declared with attribute warn_unused_result [-Wunused-result]
         fscanf(fp,"version %d npes %d ndims %d\n",&rversno, &rnpes, ndims);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../../src/clib/pioc_support.c:880:13: warning: ignoring return value of ‘fscanf’, declared with attribute warn_unused_result [-Wunused-result]
             fscanf(fp,"%d ", tdims + i);
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../../src/clib/pioc_support.c:887:13: warning: ignoring return value of ‘fscanf’, declared with attribute warn_unused_result [-Wunused-result]
             fscanf(fp, "%d %lld", &j, &maplen);
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../../src/clib/pioc_support.c:893:17: warning: ignoring return value of ‘fscanf’, declared with attribute warn_unused_result [-Wunused-result]
                 fscanf(fp, "%lld ", tmap+j);
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/Makefile.am
+++ b/Makefile.am
@@ -1,4 +1,8 @@
+# This is part of PIO. It creates the main Makefile.
+
+# Ed Hartnett
 
 SUBDIRS = src tests
-#Recommended by libtoolize 
+
+# Look in the m4 directory for autotools stuff.
 ACLOCAL_AMFLAGS= -I m4

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1,1 +1,5 @@
+# This is part of PIO. It creates the Makefile for the src directory.
+
+# Ed Hartnett
+
 SUBDIRS = clib

--- a/src/clib/Makefile.am
+++ b/src/clib/Makefile.am
@@ -4,11 +4,16 @@
 # The library we are building.
 lib_LTLIBRARIES = libpio.la
 
+# These linker flags specify libtool version info.
+# See http://www.gnu.org/software/libtool/manual/libtool.html#Libtool-versioning
+# for information regarding incrementing `-version-info`.
+libpio_la_LDFLAGS = -version-info 1:0:0
+
 # The header file.
 include_HEADERS = pio.h
 
-# THe soure files.
-libpio_la_SOURCES = bget.c pioc_sc.c pio_darray.c pio_file.c \
-pio_getput_int.c pio_msg.c pio_nc.c pio_rearrange.c pio_varm.c \
-pioc.c pioc_support.c pio_darray_int.c pio_get_nc.c pio_lists.c \
-pio_nc4.c pio_put_nc.c pio_spmd.c pio_internal.h bget.h
+# The library soure files.
+libpio_la_SOURCES = bget.c pioc_sc.c pio_darray.c pio_file.c		\
+pio_getput_int.c pio_msg.c pio_nc.c pio_rearrange.c pio_varm.c pioc.c	\
+pioc_support.c pio_darray_int.c pio_get_nc.c pio_lists.c pio_nc4.c	\
+pio_put_nc.c pio_spmd.c pio_internal.h bget.h

--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -1,1 +1,6 @@
+# This file is part of PIO. It generates the Makefiles for the tests
+# directory.
+
+# Ed Hartnett
+
 SUBDIRS = cunit

--- a/tests/cunit/Makefile.am
+++ b/tests/cunit/Makefile.am
@@ -1,28 +1,44 @@
 ## This is the automake file for building the C tests for the PIO
 ## library.
+
 # Ed Hartnett 8/17/17
 
 # Link to our assembled library.
 AM_LDFLAGS = ${top_builddir}/src/clib/libpio.la
 AM_CPPFLAGS = -I$(top_srcdir)/src/clib
 
-# The tests that will be run.
-PIO_TESTS = test_async_mpi test_spmd test_intercomm2			\
-test_async_simple test_async_3proc test_async_4proc			\
-test_iosystem2_simple test_iosystem2_simple2 test_iosystem2		\
-test_iosystem3_simple test_iosystem3_simple2 test_iosystem3 test_pioc	\
-test_pioc_unlim test_pioc_putget test_pioc_fill test_darray		\
-test_darray_multi test_darray_multivar test_darray_multivar2		\
-test_darray_multivar3 test_darray_1d test_darray_3d			\
-test_decomp_uneven test_decomps test_rearr test_darray_async_simple	\
-test_darray_async test_darray_async_many test_darray_2sync		\
-test_async_multicomp test_async_multi2 test_async_manyproc
+
+# Build the tests for make check.
+#test_intercomm2
+check_PROGRAMS = test_async_mpi test_spmd test_async_simple		\
+test_async_3proc test_async_4proc test_iosystem2_simple			\
+test_iosystem2_simple2 test_iosystem2 test_iosystem3_simple		\
+test_iosystem3_simple2 test_iosystem3 test_pioc test_pioc_unlim		\
+test_pioc_putget test_pioc_fill test_darray test_darray_multi		\
+test_darray_multivar test_darray_multivar2 test_darray_multivar3	\
+test_darray_1d test_darray_3d test_decomp_uneven test_decomps		\
+test_rearr test_darray_async_simple test_darray_async			\
+test_darray_async_many test_darray_2sync test_async_multicomp		\
+test_async_multi2 test_async_manyproc
+
+# Tests will run from a bash script.
+TESTS = run_tests.sh
+
+# # The tests that will be run.
+# PIO_TESTS = test_async_mpi test_spmd test_intercomm2			\
+# test_async_simple test_async_3proc test_async_4proc			\
+# test_iosystem2_simple test_iosystem2_simple2 test_iosystem2		\
+# test_iosystem3_simple test_iosystem3_simple2 test_iosystem3 test_pioc	\
+# test_pioc_unlim test_pioc_putget test_pioc_fill test_darray		\
+# test_darray_multi test_darray_multivar test_darray_multivar2		\
+# test_darray_multivar3 test_darray_1d test_darray_3d			\
+# test_decomp_uneven test_decomps test_rearr test_darray_async_simple	\
+# test_darray_async test_darray_async_many test_darray_2sync		\
+# test_async_multicomp test_async_multi2 test_async_manyproc
 
 # Source code for each test.
-test_async_mpi_SOURCES = test_async_mpi.c
-test_intercomm2_SOURCES = test_intercomm2.c test_common.c pio_tests.h
+#test_intercomm2_SOURCES = test_intercomm2.c test_common.c pio_tests.h
 test_async_simple_SOURCES = test_async_simple.c test_common.c pio_tests.h
-test_async_3proc_SOURCES = test_async_3proc.c test_common.c pio_tests.h
 test_async_4proc_SOURCES = test_async_4proc.c test_common.c pio_tests.h
 test_iosystem2_simple_SOURCES = test_iosystem2_simple.c test_common.c pio_tests.h
 test_iosystem2_simple2_SOURCES = test_iosystem2_simple2.c test_common.c pio_tests.h
@@ -49,23 +65,10 @@ test_darray_async_SOURCES = test_darray_async.c test_common.c pio_tests.h
 test_darray_async_many_SOURCES = test_darray_async_many.c test_common.c pio_tests.h
 test_darray_2sync_SOURCES = test_darray_2sync.c test_common.c pio_tests.h
 test_spmd_SOURCES = test_spmd.c test_common.c pio_tests.h
-test_intercomm2_SOURCES = test_async_simple.c test_common.c pio_tests.h
 test_async_3proc_SOURCES = test_async_3proc.c test_common.c pio_tests.h
 test_async_multicomp_SOURCES = test_async_multicomp.c test_common.c pio_tests.h
 test_async_multi2_SOURCES = test_async_multi2.c test_common.c pio_tests.h
 test_async_manyproc_SOURCES = test_async_manyproc.c test_common.c pio_tests.h
-
-# Build the tests for the tests target.
-tests: ${PIO_TESTS}
-
-# Build the tests for make check.
-check_PROGRAMS = $(PIO_TESTS)
-
-# Tests will run from a bash script.
-TESTS = run_tests.sh
-
-# Bash script needs all tests built.
-run_tests.sh : tests
 
 # Distribute the test script.
 EXTRA_DIST = run_tests.sh

--- a/tests/cunit/Makefile.am
+++ b/tests/cunit/Makefile.am
@@ -71,4 +71,4 @@ run_tests.sh : tests
 EXTRA_DIST = run_tests.sh
 
 # Clean up files produced during testing.
-CLEANFILES = *.nc *.log
+CLEANFILES = *.nc *.log decomp*.txt

--- a/tests/cunit/Makefile.am
+++ b/tests/cunit/Makefile.am
@@ -23,18 +23,6 @@ test_async_multicomp test_async_multi2 test_async_manyproc
 # Tests will run from a bash script.
 TESTS = run_tests.sh
 
-# # The tests that will be run.
-# PIO_TESTS = test_async_mpi test_spmd test_intercomm2			\
-# test_async_simple test_async_3proc test_async_4proc			\
-# test_iosystem2_simple test_iosystem2_simple2 test_iosystem2		\
-# test_iosystem3_simple test_iosystem3_simple2 test_iosystem3 test_pioc	\
-# test_pioc_unlim test_pioc_putget test_pioc_fill test_darray		\
-# test_darray_multi test_darray_multivar test_darray_multivar2		\
-# test_darray_multivar3 test_darray_1d test_darray_3d			\
-# test_decomp_uneven test_decomps test_rearr test_darray_async_simple	\
-# test_darray_async test_darray_async_many test_darray_2sync		\
-# test_async_multicomp test_async_multi2 test_async_manyproc
-
 # Source code for each test.
 test_intercomm2_SOURCES = test_intercomm2.c test_common.c pio_tests.h
 test_async_simple_SOURCES = test_async_simple.c test_common.c pio_tests.h

--- a/tests/cunit/Makefile.am
+++ b/tests/cunit/Makefile.am
@@ -9,17 +9,16 @@ AM_CPPFLAGS = -I$(top_srcdir)/src/clib
 
 
 # Build the tests for make check.
-#test_intercomm2
-check_PROGRAMS = test_async_mpi test_spmd test_async_simple		\
-test_async_3proc test_async_4proc test_iosystem2_simple			\
-test_iosystem2_simple2 test_iosystem2 test_iosystem3_simple		\
-test_iosystem3_simple2 test_iosystem3 test_pioc test_pioc_unlim		\
-test_pioc_putget test_pioc_fill test_darray test_darray_multi		\
-test_darray_multivar test_darray_multivar2 test_darray_multivar3	\
-test_darray_1d test_darray_3d test_decomp_uneven test_decomps		\
-test_rearr test_darray_async_simple test_darray_async			\
-test_darray_async_many test_darray_2sync test_async_multicomp		\
-test_async_multi2 test_async_manyproc
+check_PROGRAMS = test_intercomm2 test_async_mpi test_spmd		\
+test_async_simple test_async_3proc test_async_4proc			\
+test_iosystem2_simple test_iosystem2_simple2 test_iosystem2		\
+test_iosystem3_simple test_iosystem3_simple2 test_iosystem3 test_pioc	\
+test_pioc_unlim test_pioc_putget test_pioc_fill test_darray		\
+test_darray_multi test_darray_multivar test_darray_multivar2		\
+test_darray_multivar3 test_darray_1d test_darray_3d			\
+test_decomp_uneven test_decomps test_rearr test_darray_async_simple	\
+test_darray_async test_darray_async_many test_darray_2sync		\
+test_async_multicomp test_async_multi2 test_async_manyproc
 
 # Tests will run from a bash script.
 TESTS = run_tests.sh
@@ -37,7 +36,7 @@ TESTS = run_tests.sh
 # test_async_multicomp test_async_multi2 test_async_manyproc
 
 # Source code for each test.
-#test_intercomm2_SOURCES = test_intercomm2.c test_common.c pio_tests.h
+test_intercomm2_SOURCES = test_intercomm2.c test_common.c pio_tests.h
 test_async_simple_SOURCES = test_async_simple.c test_common.c pio_tests.h
 test_async_4proc_SOURCES = test_async_4proc.c test_common.c pio_tests.h
 test_iosystem2_simple_SOURCES = test_iosystem2_simple.c test_common.c pio_tests.h

--- a/tests/cunit/run_tests.sh
+++ b/tests/cunit/run_tests.sh
@@ -9,7 +9,8 @@ trap exit SIGINT SIGTERM
 
 printf 'running PIO tests...\n'
 
-PIO_TESTS='test_async_mpi test_spmd test_rearr test_intercomm2 test_async_simple '\
+# test_intercomm2
+PIO_TESTS='test_async_mpi test_spmd test_rearr test_async_simple '\
 'test_async_3proc test_async_4proc test_iosystem2_simple test_iosystem2_simple2 '\
 'test_iosystem2 test_iosystem3_simple test_iosystem3_simple2 test_iosystem3 test_pioc '\
 'test_pioc_unlim test_pioc_putget test_pioc_fill test_darray test_darray_multi '\

--- a/tests/cunit/run_tests.sh
+++ b/tests/cunit/run_tests.sh
@@ -9,8 +9,7 @@ trap exit SIGINT SIGTERM
 
 printf 'running PIO tests...\n'
 
-# test_intercomm2
-PIO_TESTS='test_async_mpi test_spmd test_rearr test_async_simple '\
+PIO_TESTS='test_intercomm2 test_async_mpi test_spmd test_rearr test_async_simple '\
 'test_async_3proc test_async_4proc test_iosystem2_simple test_iosystem2_simple2 '\
 'test_iosystem2 test_iosystem3_simple test_iosystem3_simple2 test_iosystem3 test_pioc '\
 'test_pioc_unlim test_pioc_putget test_pioc_fill test_darray test_darray_multi '\

--- a/tests/cunit/test_intercomm2.c
+++ b/tests/cunit/test_intercomm2.c
@@ -9,6 +9,7 @@
  * @author Ed Hartnett
  *
  */
+#include <config.h>
 #include <pio.h>
 #include <pio_tests.h>
 #include <pio_internal.h>

